docs/cli/ilab-processes.md

```diff
@@ -0,0 +1,53 @@
+# Processes in InstructLab
+
+The ability to detach from processes is crucial to the user experience of InstructLab. However, the concept of multi-processing, process management, and the monitoring of processes is very complex.
+
+It is important to try and add this concept in as simply as possible, expanding on the state reporting, logging, and other features as we go along.
+
+## Phased approach to InstructLab Processes
+
+This document is going to describe phase 1 of implementing processes in InstructLab. Phase 1 is to be described as the "ilab simple process management system". This will depend purely on python packages, PID tracking, and log files to create the experience of detachable processes. The key here is the concept of the UUID, allowing a future REST API to keep track of InstructLab processes using these unique identifiers. 
+
+We can re-visit all this in phase 2, when we discuss if we want to utilize something lils systemd or a more in-depth process-monitor repo to track processes.
+
+### Phase 1
+
+Phase one would focus on adding the concept of detaching from processes. This would be implemented in `ilab 0.23.0`.
+
+The workflow would allow for:
+
+`ilab data generate -dt` (run a detached generation process)
+`ilab model train -dt` (run a detached training process)
+
+`ilab process list`
+
+```console=
++----------------------------------------+------------+-------------+---------+
+| Proccess UUID                          | Type       | Duration    | Status  |
++----------------------------------------+--------------------------+---------+
+| train-2423813fhthfbnaj                 | training   | 8 hr 32 min | Running |
+| generation-6768769599sdbjklm           | generation | 3 hr 15 min | Running |
+| generation-1345661hlsmma               | generation | 22 min      | Errored |
++----------------------------------------+---------------------+--------------+
+
+```
+
+`ilab process attach <UUID>`
+
+This command would re-attach to the given process, allowing to user to view the live logs of the process.
+
+These commands will be done in a very simple way at first using the following architecture:
+
+1. The backbone of this process mangement will be the python multiprocessing package, using `Process` to start the detached commands and `Manager` to maintain an in-code dictionary of the existing processes.
+2. a detached process be re-attachable by tailing the log file and then allowing the user to ctrl+c the process as normal using `KeyboardInterrupt`
+3. A `Manager.dict` process registry will be maintained for tracking UUIDs created via the `uuid` python package, the PID of the actual process, and a `log_file` where the process will be outputting its logs to so that the user can re-attach. this log file dir will be tracked using our `DEFAULTS` package and will be standard throughout releases.
+    
+The general flow would be:
+
+1. a user runs `ilab data generate -dt`
```

Comment:
Which processes explicitly would get "detachablility" functionality? can we have an explicit list as part of this doc?

Reply:
probably the longest running ones to begin with at least
download, sdg, train